### DISCUSSION TITLE
event data

### DISCUSSION THREAD

**Discussion by chanansh** [blue-yonder/tsfresh#878] [Q&A]
Does the package supports event data rather than time-series data, e.g. `event_id, event_time, event_type` ?

**Comment by nils-braun** [comment]
How would you define "event data"? What is the difference between event data and time series?
In principle, the tsfresh framework is agnostic to what your data describe - it might be the case that some of the feature extractors do not make sense (but this van be the case anyway, not only for event data, but for all kinds of data).

**Reply by kempa-liehr** [reply]
Just adding to the answer from @nils-braun. In general, tsfresh works on all kinds of ordered data. In many applications, the ordering dimension will be time (time-series), other commonly found ordering dimensions are wavelength (e.g reflectance spectra) or a spatial dimension. These examples have in common that the input signal is sampled regularly (the differences of successive coordinates of the ordering dimension are constant).  As @nils-braun  pointed out, the extracted features can be directly interpreted.

Your example refers to event sequences, which are typically found in process monitoring. The ordering dimension of event sequences is time, but these event sequences are not regularly sampled. If a process model is available, event sequences can be modelled as time-continuous [process evolution functions](https://doi.org/10.1186/s40537-015-0017-0) from which a time-series can be sampled.
However, features can also be extracted directly from event sequences, like e.g. [ sequences of word lengths](https://doi.org/10.1140/epjds/s13688-020-00244-9), but the interpretation will be more difficult.

**Comment by chanansh** [comment]
The main issue of event data is that it is NOT regularly sampled and very
sparse
